src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -422,12 +422,13 @@ private function resolveVariables($value)
 
         $regex = '/
             (?<!\\\\)
-            (?P<backslashes>\\\\*)             # escaped with a backslash?
+            (?P<backslashes>\\\\*)                # escaped with a backslash?
             \$
-            (?!\()                             # no opening parenthesis
-            (?P<opening_brace>\{)?             # optional brace
-            (?P<name>'.self::VARNAME_REGEX.')? # var name
-            (?P<closing_brace>\})?             # optional closing brace
+            (?!\()                                # no opening parenthesis
+            (?P<opening_brace>\{)?                # optional brace
+            (?P<name>'.self::VARNAME_REGEX.')?    # var name
+            (?P<default_value>(?:(:-)([^\}]+)))?  # optional default value
```

Comment:
no need for the extra brackets I believe: `(?P<default_value>:-[^\}]+)?`
but more generally, the parsing logic doesn't handle quoted/interpolated values
eg. `${ABC:-a\'aa}` or worse `${ABC:-a\'a{a}a}` (which results in `a'a{aa}`)
an idea could be to not deal with these and reject more chars than only `}`?

Reply:
Sounds good to not deal with these cases. It's a fallback mechanism which should contain simple values. Let's keep it KISS for now. 

I will update the regex. And should we reject `}{\'` characters with a FormatException?

Reply:
I think so yes. I would also exclude double quotes and `$`, and maybe more special chars (I don't know them all)?

Reply:
@nicolas-grekas I removed the braces from the regex. The regex strips only the last `}` from the default value and checks later with `strpbrk('"{}$)` for the invalid characters. This way we can throw a FormatException instead of simply ignoring the default value.

I hope my solution is okay, as my knowledge of regexes is not great. Let me know what you think.